### DISCUSSION TITLE
Albums with many photos

### DISCUSSION THREAD

**Discussion by tinohager** [LycheeOrg/Lychee#2721] [General]
I have created a new album with 340 photos. When I open the album, I see a blank page without any information about the loading process, no menu, just a white page.

- `api/v2/Album?album_id=N-XXXXXXXXXXXX -> loading time is `1.5s` (1MB)`
- Time to view the album content `3s`

In the JSON response, I noticed a few points worth optimizing:

- `next_photo_id` and `previous_photo_id`: These values can likely be calculated on the client side, reducing server-side overhead.
- `album_id`: Could this field be removed entirely if it's redundant?
- Empty JSON fields: Removing fields that are empty on the server side could further streamline the response.
- `size_variants` in URLs: There’s potential to save space by optimizing how these URLs are structured, as they remain largely identical.

**Comment by ildyria** [comment]
the next and previous ids may be able to be dropped indeed. 
This is legacy from v3. I haven't really thought that part again.

There was a reason for giving `album_id`, I am not sure which one anymore.

The size_variants, this is pretty much automated from the DTO, the underlying library (spatie/Laravel-data) and the relationship implementation. I am not sure there is much I can do here.

Possibly adding pagination & infinite scrolling would be a nicer approach (though the coming back from photo view and reloading is going to be tricky).

**Comment by tinohager** [comment]
In my opinion, the router switches to the desired page, triggering the request to load the album information. Normally, you could use a variable to indicate the loading state and show a loading bar. Once the data is available, the loading bar can be hidden.
